src/Tasks/AssemblyDependency/ReferenceTable.cs

```diff
@@ -2687,6 +2687,15 @@ internal void GetReferenceItems
             copyLocalFiles = copyLocalItems.ToArray();
         }
 
+        internal static ITaskItem[] NormalizeAsArray(List<ITaskItem> items)
+        {
+            return items.Select(ti =>
+            {
+                ti.ItemSpec = FileUtilities.NormalizePath(ti.ItemSpec);
+                return ti;
+            }).ToArray();
```

Comment:
I'd try to avoid unnecessary allocations here as it's a hot path. I'd pre-allocate the final array, and the `for` over each item and fill it in (to even avoid the allocation for the foreach iterator).

Reply:
Alternatively, an IEqualityComparer<T> that ignores trailing directory separators canbe used.

Reply:
That would miss that C:\\\\\\foo is the same as C:\foo